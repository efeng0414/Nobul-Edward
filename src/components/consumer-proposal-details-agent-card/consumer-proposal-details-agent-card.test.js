import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";

import ConsumerProposalDetailsAgentCardWrapped from "./index";

describe("ConsumerProposalDetailsAgentCard", () => {
  const {
    WrappedComponent: ConsumerProposalDetailsAgentCard
  } = ConsumerProposalDetailsAgentCardWrapped;

  const componentProps = {
    intl: global.getIntl(),
    agentAvatar: "Avatar",
    agentName: "John Smith",
    agentOffice: "REMAX",
    agentAbout: "This is some text about the agent"
  };

  const componentJSX = <ConsumerProposalDetailsAgentCard {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.intl}>{componentJSX}</IntlProvider>
    );

    const componentSnapshot = renderer.create(snapshotJSX).toJSON();
    it("should matched stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Render method", () => {
    const wrapper = global.Enzyme.mount(componentJSX);
    it("should render the card body ", () => {
      expect(wrapper.find(".agent-card__body").length).toEqual(1);
    });

    it("should render the card footer ", () => {
      expect(wrapper.find(".agent-card__footer").length).toEqual(1);
    });
  });
});
