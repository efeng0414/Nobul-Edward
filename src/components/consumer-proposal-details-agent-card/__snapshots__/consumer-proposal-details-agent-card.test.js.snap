// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerProposalDetailsAgentCard Component Snapshot should matched stored snapshot 1`] = `
<div
  className="agent-card"
>
  <div
    className="agent-card__body"
  >
    <div
      className="agent-card__avatar"
    >
      <span
        className="ant-avatar ant-avatar-circle ant-avatar-image"
        style={
          Object {
            "fontSize": 18,
            "height": 150,
            "lineHeight": "150px",
            "width": 150,
          }
        }
      >
        <img
          onError={[Function]}
          src="Avatar"
        />
      </span>
    </div>
    <div
      className="agent-card__agent-name"
    >
      John Smith
    </div>
    <div
      className="agent-card__agent-office"
    >
      REMAX
    </div>
    <div
      className="agent-card__agent-about"
    >
      This is some text about the agent
    </div>
  </div>
  <div
    className="agent-card__footer"
  >
    <div
      className="footer__icon"
    >
      <div
        className="footer__image-container"
      >
        <img
          alt="Phone Icon"
          className="footer__image"
          src="file-mock"
        />
      </div>
    </div>
    <div
      className="footer__action-content"
    >
      <div
        className="footer__title"
      >
        Contact the Agent
      </div>
      <div
        className="footer__text"
      >
        Set up a meeting to interview the agent before you decide to work with them.
      </div>
    </div>
  </div>
</div>
`;
